File: golang/client/commands/removeClientCommandHandler.go
package client

import (
	client "github.com/houstondapaz/cqrs-govscsharp/golang/client/domain"
	commands "github.com/houstondapaz/cqrs-govscsharp/golang/domain/commands"
	events "github.com/houstondapaz/cqrs-govscsharp/golang/domain/events"
)

type RemoveClientCommandHandler struct {
	commands.CommandHandler
	storage  client.ClientStorager
	eventBus events.EventBuser
}

func NewRemoveClientCommandHandler(clientStorager client.ClientStorager, eventBus events.EventBuser) RemoveClientCommandHandler {
	return RemoveClientCommandHandler{
		storage:  clientStorager,
		eventBus: eventBus,
	}
}

func (ch RemoveClientCommandHandler) CommandName() string {
	return "RemoveClientCommand"
}

func (ch RemoveClientCommandHandler) Handle(command commands.Commander) error {
	c, err := ch.storage.GetOne(command.(client.RemoveClientCommand).ClientId())
	if err != nil {
		return err
	}

	c.Disable()

	err = ch.storage.Update(c)

	if err != nil {
		return err
	}

	return ch.eventBus.Publish(c.PendingEvents()...)
}

File: csharp/src/CQRSService.Client/Commands/RemoveClientCommandHandler.cs
using CQRSService.Client.CQRSService.Domain.Commands;
using CQRSService.Client.Storage;
using CQRSService.Domain.Commands;
using CQRSService.Domain.Events;
using MediatR;
using System.Threading;
using System.Threading.Tasks;

namespace CQRSService.Client.Commands
{
    public class RemoveClientCommandHandler : ICommandHandler<RemoveClientCommand>
    {
        private readonly IStorage<Client> _clientStorage;
        private readonly IEventBus _eventBus;

        public RemoveClientCommandHandler(IStorage<Client> storageClient, IEventBus eventBus)
        {
            _clientStorage = storageClient;
            _eventBus = eventBus;
        }
        public async Task<Unit> Handle(RemoveClientCommand request, CancellationToken cancellationToken)
        {
            var client = await _clientStorage.GetOne(request.Id);
            client.Disable();

            await _clientStorage.Update(client);
            await _eventBus.Publish(client.PendingEvents.ToArray());

            return Unit.Value;
        }
    }
}
